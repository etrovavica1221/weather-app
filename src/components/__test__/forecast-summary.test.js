import "@testing-library/jest-dom/extend-expect";
import React from 'react';
import { render } from "@testing-library/react";
import ForecastSummary from '../forecast-summary.js';

describe("ForecastSummary", () => {
    it('renders correctly', () => {
       const { asFragment } = render(
        <ForecastSummary
            date={1525305600000}
            temperature="20&deg;c"
            description="Tornado"
            icon="781"
         />
       );
       expect(asFragment()).toMatchSnapshot();
    });

    it("renders the correct props", () => {
        const { getByText, getByTestId } = render(
            <ForecastSummary
                date={1525305600000}
                temperature="20&deg;c"
                description="Tornado"
                icon="781"
             />
        );

        expect(getByTestId("date-id")).toHaveClass("date");
        expect(getByTestId("temperature-id")).toHaveClass("temperature");
        expect(getByText("Tornado")).toHaveClass("description");
        expect(getByTestId("icon-id")).toHaveClass("icon");
    });
});