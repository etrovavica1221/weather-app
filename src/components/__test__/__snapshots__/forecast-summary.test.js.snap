// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <span
    class="forecast-summary"
  >
    <div
      class="date"
      data-testid="date-id"
    >
      Thu 3rd May
    </div>
    <div
      class="temperature"
      data-testid="temperature-id"
    >
      20
    </div>
    <div
      class="description"
      data-testid="description-id"
    >
      Tornado
    </div>
    <div
      class="icon"
      data-testid="icon-id"
    >
      <i
        class="wi wi-tornado wi-flip-horizontal wi-rotate-90"
        data-testid="testIcon"
        name="owm"
      />
    </div>
  </span>
</DocumentFragment>
`;
